br/pkg/storage/azblob.go

```diff
@@ -324,6 +350,35 @@ type AzureBlobStorage struct {
 
 	cpkScope *blob.CPKScopeInfo
 	cpkInfo  *blob.CPKInfo
+
+	// resolvedAccountName is the final account name we are going to use.
+	resolvedAccountName     string
+	resolvedServiceEndpoint string
+}
+
+// CopyFrom implements Copier.
+func (s *AzureBlobStorage) CopyFrom(ctx context.Context, e ExternalStorage, spec CopySpec) error {
+	es, ok := e.(*AzureBlobStorage)
+	if !ok {
+		return errors.Annotatef(berrors.ErrStorageInvalidConfig,
+			"AzureBlobStorage.CopyFrom supports *AzureBlobStorage only, got %T", e)
+	}
+
+	url, err := urlOfObjectByEndpoint(es.resolvedServiceEndpoint, es.options.Bucket, es.withPrefix(spec.From))
+	if err != nil {
+		return errors.Annotatef(err, "failed to get url of object %s", spec.From)
+	}
+	dstBlob := s.containerClient.NewBlobClient(s.withPrefix(spec.To))
+
+	// NOTE: `CopyFromURL` supports files up to 256 MiB, which might not be enough for huger regions?
```

Comment:
a good catch. if the error code returned does not contain the detailed reason, we have to document the restriction in release notes.